src/librustc_typeck/diagnostics.rs

```diff
@@ -1611,9 +1611,9 @@ static BAR: _ = "test"; // error, explicitly write out the type instead
 "##,
 
 E0122: r##"
-An attempt was made to add a generic constraint to a type alias. While Rust will
-allow this with a warning, it will not currently enforce the constraint.
-Consider the example below:
+An attempt was made to add a generic constraint to a type alias. This constraint
+is entirely ignored. For backwards compatibility, Rust still allows this with a
+warning. Consider the example below:
```

Comment:
This is incorrect. The constraint is required to use associated type syntax (without writing `<T as Trait>::Assoc` - which you can't for `Fn` traits on stable).